### PR TITLE
Improved metronome (on/off during song, pattern and bb playback)

### PR DESCRIPTION
**Pull request by michaelgregorius**
There is a new tool button that can be used to turn the metronome on and
off. Per default the metronome is turned off. When enabled the metronome
will play during on song playback, pattern playback and BB playback. During
export it is ignored.

A new icon was added as well.

The state is currently stored in the `Mixer`. It might make sense to put
the metronome configuration in its own class in the future. The state is
currently not stored in the file but this might be a good choice for now
until a better place is found for the metronome data.

### CODE REVIEW

**Comment by michaelgregorius** [comment]
I saw in the forum that there seems to be a need for an improved metronome functionality so I thought to give it a first shot. :)
metronome-icon

Also related to #475.

**Comment by tresf** [comment]
Well need addition!  

Also related:  #869 (@musikBear), #1357 (@Reaper10)

File: src/core/Mixer.cpp [hunk 1]
```diff
@@ -318,10 +319,13 @@ const surroundSampleFrame * Mixer::renderNextBuffer()
...
-		gui->pianoRoll()->isRecording() == true &&
+	Song::PlayModes currentPlayMode = Engine::getSong()->playMode();
+	Song::PlayPos p = Engine::getSong()->getPlayPos( currentPlayMode );
+
+	if( (   currentPlayMode == Song::Mode_PlayPattern || currentPlayMode == Song::Mode_PlaySong
+	     || currentPlayMode == Song::Mode_PlayBB ) &&
+		m_metronomeActive &&
+		!Engine::getSong()->isExporting() &&
 		p != last_metro_pos )
 	{
```
Comment (Wallacoloo) [inline src/core/Mixer.cpp]: What do you think of decomposing the conditional a bit? Maybe something like:
```c++
bool isPlaying = currentPlayMode == Song::Mode_PlayPattern || currentPlayMode == Song::Mode_PlaySong
	     || currentPlayMode == Song::Mode_PlayBB;
if (isPlaying && m_metronomeActive && !Engine::getSong()->isExporting() &&
 		p != last_metro_pos) ...
```

Hope I'm not being too pedantic. Everything else looks good.

**Comment by Wallacoloo** [comment]
Tested & works as described (via `git checkout michaelgregorius/metronome-improvements && git rebase master && make install`).

I've left one code comment for @michaelgregorius to address. Pending further feedback, this should be good to merge.

**Comment by musikBear** [comment]
@michaelgregorius Big :+1: The metronome has been one of the most criticized features, and a toggle options is brilliant. The Master's two different sounds, and yours toggle-option, makes metronome a really good feature in lmms!

**Comment by michaelgregorius** [comment]
@Wallacoloo I have incorporated your proposal for a decomposition of the conditional. I have also rebased against master and squashed the commit.

**Comment by michaelgregorius** [comment]
@musikBear Thanks for the kind words! Knowing that a feature is wanted by users and therefore is also used is always a great motivation. :smiley:

**Comment by Wallacoloo** [comment]
:+1:

**Pull request merged**